tests/unit/stream/test_downloader.py

```diff
@@ -80,3 +84,17 @@ def _mock_accumulate_peers(q1, q2):
             self.assertEqual(f.read(), self.stream_bytes)
         # self.assertIs(self.server_from_client.tcp_last_down, None)
         # self.assertIsNot(bad_peer.tcp_last_down, None)
+
+    async def test_client_chunked_response(self):
+        self.server.stop_server()
+        class ChunkedServerProtocol(BlobServerProtocol):
+
+            def send_response(self, responses):
+                to_send = []
+                while responses:
+                    to_send.append(responses.pop())
+                for byte in BlobResponse(to_send).serialize():
+                    self.transport.write(bytes([byte]))
+        self.server.server_protocol_class = ChunkedServerProtocol
+        self.server.start_server(33333, '127.0.0.1')
+        await self._test_transfer_stream(10)
```

Comment:
Can this test have an assert somewhere?

Reply:
_test_transfer_stream does it, but I will carry some to the test itself.
What's being really tested here is that it doesnt hang or explodes in errors (which was the fix). I will assert it transferred fine.

Reply:
For hang, is it possible to assert that it finishes in 1s or 2s, etc (and fail test if it takes longer)?